### ISSUE TITLE
Storage box for new dataset created by MyTardis web interface

### ISSUE THREAD

**Issue by wettenhj** [mytardis/mytardis#370]
[The first paragraph may no longer be relevant - because the dataset model no longer has a storage box field.]
I have two storage boxes - local and staging.  When I create a dataset in MyTardis's web interface (as distinct from the RESTful API), there is a select box on the "add-dataset" form where I can choose the "local" storage box, but my choice seems to be ignored - I'm currently finding that new datasets are defaulting to "staging".  I can't see any reference to the storage box in the "add_dataset" method in views.py: https://github.com/mytardis/mytardis/blob/develop/tardis/tardis_portal/views.py#L3219 

I'm wondering about the reason for the StorageBox class's get_default_storage_box returning the "first" storage box, and what that actually means.  It currently seems to return StorageBox.objects.all()[0].  Would it make more sense to return StorageBox.objects.all().order_by('id')[0] ?

```
>>> StorageBox.objects.all()
[<StorageBox: staging>, <StorageBox: local>]
>>> StorageBox.objects.all().order_by('id')
[<StorageBox: local>, <StorageBox: staging>]
```

**Comment by wettenhj** [comment]
This issue may have been created back when the dataset model still had a storage box field, which is doesn't any more.  But the issue of the arbitrary storage box returned by get_default_storage_box still remains.  Perhaps each storage box could be given a priority, so the default storage box is the one with the highest priority, instead of the one at index zero?

**Comment by wettenhj** [comment]
https://github.com/mytardis/mytardis/blob/develop/tardis/tardis_portal/models/storage.py#L151

**Comment by grischa** [comment]
I think priority is not very meaningful.
How about one of these solutions:

1. lowest id storage box is default
2. Have a `StorageBoxAttribute`: `key='default'`, `value=True`
3. find a storage box where location is `DEFAULT_STORAGE_BASE_DIR`
4. create a default storage box at `DEFAULT_STORAGE_BASE_DIR`
5. a combination of them with, i.e. try first 2, then 3, then 1, then 4

**Comment by jasonrig** [comment]
Regarding option 5: should probably be 2, 3, 4, 1. Option 1 will always work if a storage box exists, but creating a default storage box according to the setting _before_ defaulting to the first available probably makes more sense.

**Comment by wettenhj** [comment]
I like option 2 best.

I suppose the possible need for 4 is related to how MyTardis is installed - whether fixtures are used to automatically create default storage box(es).

I do have a slight concern that if you provide multiple ways to do the same thing, then some people will use both methods concurrently (settings.py and StorageBoxAttribute) which then means multiple places to update if it ever needs updating.  But I'm not strongly against supporting both.

**Comment by grischa** [comment]
closed by #578
